File: javascript-oop/prototype-students/person.js
(function () {
    var person = (function () {
        var person = {},
            _name,
            _age;

        Object.defineProperties(person, {
            init: {
                value: function (name, age) {
                    this.name = name;
                    this.age = age;
                    return this;
                }
            },
            name: {
                get: function () {
                    return _name;
                },
                set: function (value) {
                    _name = value;
                }
            },
            age: {
                get: function () {
                    return _age;
                },
                set: function (value) {
                    _age = value;
                }
            }
        });

        return person;
    } ());

    module.exports = person;
} ());

File: HQC-Part-1/homework-02-Naming-Identifier/GardenOfEden/GardenOfEden/Models/Person.cs
using GardenOfEden.Enums;

namespace GardenOfEden.Models
{
    public class Person
    {
        private Person()
        {
        }

        public GenderType Gender { get; set; }

        public string Name { get; set; }

        public int Age { get; set; }

        public static Person Create(int age)
        {
            var newPerson = new Person();

            newPerson.Age = age;
            if (age % 2 == 0)
            {
                newPerson.Name = "Dude";
                newPerson.Gender = GenderType.Male;
            }
            else
            {
                newPerson.Name = "Dudette";
                newPerson.Gender = GenderType.Female;
            }

            return newPerson;
        }
    }
}